---
title: "microbiomeDataSets"
date: "`r Sys.Date()`"
package: microbiomeDataSets
output: 
  BiocStyle::html_document:
    fig_height: 7
    fig_width: 10
    toc: yes
    toc_depth: 2
    number_sections: true
vignette: >
  %\VignetteIndexEntry{microbiomeDataSets}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: references.bib  
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(microbiomeDataSets)
```

# Microbiome example data sets

The data sets are primarily named by the first author of the
associated publication, together with a descriptive suffix. Aliases
are provided for some of the data sets.

A table of the available data sets is available through the `availableDataSets`
function.

```{r}
availableDataSets()
```

The data is usually loaded as a `r Biocpkg("TreeSummarizedExperiment")` or a
`r Biocpkg("MultiAssayExperiment")`, if microbiome associated data was used
in the respective study. For more information on how to use these objects,
please refer to the vignettes of the packages.

All data are downloaded from ExperimentHub and cached for local re-use.

## Hintikka XO data

The Hintikka XO data contains microbiota, metabolite and biomarker
data from [Hintikka et al. (2021)](https://doi.org/10.3390/ijerph18084049) [@hintikka2021], including high-throughput profiling
data from 40 rat samples, including 39 biomarkers, 38 metabolites
(NMR), and 12706 OTUs from 318 species, measured from Cecum. This is
diet comparison study with High/Low fat diet and xylo-oligosaccaride
supplementation.

The data set is provided as a MultiAssayExperiment.


## HIT Chip Data

The following data sets are based on the Human Intestinal Tract (HIT)Chip
phylogenetic 16S microarray [(Rajilić-Stojanović _et al._
2009)](https://doi.org/10.1111/j.1462-2920.2009.01900.x). This profiling
technology differs from the more widely used 16S rRNA amplicon sequencing.

Since the data is also available in phyloseq format through the microbiome R
package and are referenced usually via aliases, they are described in more
detail.

### Intestinal microbiota profiling of 1006 Western adults

This data set from
[Lahti et al. Nat. Comm. 5:4344, 2014](https://doi.org/10.1038/ncomms5344)
comes with 130 genus-like taxonomic groups across 1006 western adults with no
reported health complications. Some subjects have also short time series.

```{r atlasdata, warning=FALSE, message=FALSE}
LahtiWAData()

# Alias
# atlas1006()
```

### Diet swap between Rural and Western populations

A two-week diet swap study between western (USA) and traditional (rural Africa)
diets, reported in
[O'Keefe et al. Nat. Comm. 6:6342, 2015](http://dx.doi.org/10.1038/ncomms7342).

```{r dietswap}
OKeefeDSData()

# Alias
# dietswap()
```

### Intestinal microbiota versus blood metabolites

Data set from [Lahti et al. PeerJ 1:e32, 2013](https://doi.org/10.7717/peerj.32)
characterizes associations between human intestinal microbiota and blood serum
lipids. Note that this data set contains an additional assay of lipid species,
and is therefore provided as `MultiAssayExperiment` object.

```{r peerj}
LahtiMLData()

# Alias
# peerj32()
```


### Version info


```{r}
sessionInfo()
```

# References
